ui/src/utils/workspace.fs.ts

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { dirname } from 'path';
+import { FileStat, FileType, Uri, workspace } from 'vscode';
+import { parseError } from '../parseError';
+
+export async function ensureFile(path: string): Promise<void> {
+    let stats: FileStat | undefined;
+    const file = Uri.file(path);
+    try {
```

Comment:
So what's your big picture plan? You kinda pushed a bunch of changes without any description/summary. Are we going to copy/paste this code to every repo or is it worth exporting some sort of `AzExtWorkspaceFs` or `AzExtFsExtra` from the UI package?

Also before we go hog wild and do this everywhere, can we do two things:
1. Actually confirm this work is worth it - I quoted the docs for reasons why I think it's better, but I haven't actually tested those features myself
2. Write some unit tests. When we're writing code that will affect every single extension, I just feel like it's a lot easier to write tests and find bugs _before_ we convert every extension. Even if we're just copying/pasting this code, I still like to have one "source" repo with tests